src/test/scala/mesosphere/marathon/storage/migration/MigrationTo19100Test.scala

```diff
@@ -0,0 +1,68 @@
+package mesosphere.marathon
+package storage.migration
+
+import akka.stream.scaladsl.{Sink, Source}
+import mesosphere.AkkaUnitTest
+import mesosphere.marathon.core.instance.Instance
+import mesosphere.marathon.core.storage.store.PersistenceStore
+import mesosphere.marathon.core.storage.store.impl.zk.{ZkId, ZkSerialized}
+import mesosphere.marathon.state.PathId
+import mesosphere.marathon.storage.repository.InstanceRepository
+import play.api.libs.json.{JsObject, JsString, Json}
+
+class MigrationTo19100Test extends AkkaUnitTest {
+
+  val mesosDefaultRole = "newDefaultRole"
+
+  "Migration to 19.100" should {
+
+    "only update instances with no role" in {
+
+      Given("an instance with and one without role")
+      val f = new Fixture()
+      val instanceId1 = Instance.Id.forRunSpec(PathId("/app"))
+      val instanceId2 = Instance.Id.forRunSpec(PathId("/app2"))
+
+      val instances = Source(List(
+        f.legacyInstanceJson(instanceId1),
```

Comment:
Is there a really good reason why we don't migrate on the protobuf directly? I'd love to understand the reasoning here better.

Reply:
ohhhhhhhhh

We store JSON, not protobuf, for our instances. 🤦‍♂ as you were.